web/src/views/Home/parts/NodeTable.tsx

```diff
@@ -0,0 +1,514 @@
+import React, { useState } from 'react';
+import get from 'lodash/get';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+
+import {
+  Box,
+  SxProps,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TablePagination,
+  TableRow,
+} from '@mui/material';
+
+import { useLimits } from 'hooks';
+import { Checkbox } from 'components';
+import {
+  FOOTER_HEIGHT,
+  HEADER_HEIGHT_DESKTOP,
+  HEADER_HEIGHT_MOBILE,
+} from 'constant';
+import { Mode } from 'enums';
+
+import SearchBar from './SearchBar';
+import ColumnsSetting from './ColumnsSetting';
+import EditSelectedNodes from './EditSelectedNodes';
+import TableSortIcon from './TableSortIcon';
+
+interface Column {
+  id: ColumnId;
+  headerStyles?: SxProps;
+  format?: (t: TFunction, nodeLimit: NodeLimit, value?: any) => string;
+  style?: (value?: any) => SxProps;
+}
+
+interface NodeTableHeaderCellProps {
+  columnGroups: ColumnId[][];
+  order: Order;
+  orderBy: ColumnId;
+  handleRequestSort: (property: ColumnId) => void;
+}
+
+const defaultCellFormat = (t: TFunction, nodeLimit: NodeLimit, value: string) =>
+  value === '0' ? '-' : value;
+const defaultCellStyle = (value: string) =>
+  value === '0' ? { color: '#CFD0DB' } : {};
+const limitStyle = (value?: string) => (value ? {} : { color: '#6E6F78' });
+
+const columns: Column[] = [
+  { id: 'alias', format: defaultCellFormat, style: defaultCellStyle },
```

Comment:
What is missing here from the previous ui is a tooltip with the node pubkey

Reply:
This logic is also useful to have: https://github.com/lightningequipment/circuitbreaker/blob/97297aa6f59d646104d7d1a977621ada69fdcbb3/webui/src/App.js#L81

Nodes may not have an alias set, and without this display logic there will be empty rows.

Reply:
I'll add it to the "Peer" cell on hover

Reply:
Didn't know that alias could be undefined! Great catch. Will add the same logic

Reply:
This also applies to the peer list in the edit dialog.